services/collectd/config.go

```diff
@@ -97,6 +107,12 @@ func (c *Config) WithDefaults() *Config {
 	if d.TypesDB == "" {
 		d.TypesDB = DefaultTypesDB
 	}
+	if d.SecurityLevel == "" {
```

Comment:
I added a Validate() function. I tested that configs with an invalid security level prevent influxd from starting. But I believe the test to assign a default value remains valid, so no change on this particular line.